src/Web/E-Games.Web/E-Games.Web/Controllers/UserController.cs

```diff
@@ -0,0 +1,108 @@
+﻿using E_Games.Data.Data.Models;
+using E_Games.Web.ViewModels;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.AspNetCore.Mvc;
+using System.Security.Claims;
+
+namespace E_Games.Web.Controllers
+{
+    [ApiController]
+    [Route("api/user")]
+    public class UserController : ControllerBase
+    {
+        private readonly UserManager<ApplicationUser> _userManager;
+
+        public UserController(UserManager<ApplicationUser> userManager)
+        {
+            _userManager = userManager;
+        }
+
+        [HttpGet]
+        [Authorize]
+        public async Task<IActionResult> UserProfile()
+        {
+            var userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+            var user = await _userManager.FindByIdAsync(userId!);
+
+            if (user == null)
+            {
+                return NotFound(new { Message = "User not found" });
+            }
+
+            var userProfile = new UserProfileModel()
+            {
+                Email = user.Email,
+                PhoneNumber = user.PhoneNumber,
+                AddressDelivery = user.AddressDelivery
+            };
+
+            return Ok(userProfile);
+        }
+
+        [HttpPut]
+        [Authorize]
+        public async Task<IActionResult> UpdateProfileAsync([FromBody] UpdateUserModel model)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest(ModelState);
+            }
+
+            var userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+            var user = await _userManager.FindByIdAsync(userId!);
+
+            if (user == null)
+            {
+                return NotFound(new { Message = "User not found" });
+            }
+
+            user.UserName = model.UserName;
+            user.PhoneNumber = model.PhoneNumber;
+            user.AddressDelivery = model.AddressDelivery;
+
+            var result = await _userManager.UpdateAsync(user);
+
+            if (!result.Succeeded)
```

Comment:
It is better to add custom exceptions and exception handling middleware to reduce this check